src/main/scala/devices/debug/SBA.scala

```diff
@@ -0,0 +1,320 @@
+
+        SBDATARdData(i) := Cat(d.reverse)
+
+        Seq(RWNotify(32, SBDATARdData(i).asUInt(), SBDATAWrData(i), SBDATARdEn(i), SBDATAWrEn(i),
+          Some(RegFieldDesc("dmi_sbdata"+i, "", reset=Some(0)))))
+      } else {
+        Seq.empty[RegField]
+      }
+    }
+
+    sb2tl.module.io.dataIn := Mux(sb2tl.module.io.wrEn,Cat(SBDATAWrData.reverse),Cat(SBDATAFieldsReg.flatten.reverse))
+    anyDataRdEn            := SBDATARdEn.reduce(_ || _)
+    anyDataWrEn            := SBDATAWrEn.reduce(_ || _)
+
+    val tryWrEn = SBDATAWrEn(0)
+    val tryRdEn = (SBADDRESSWrEn(0) && SBCSFieldsReg.sbreadonaddr) || (SBDATARdEn(0) && SBCSFieldsReg.sbreadondata)
+
+    val sbAccessError = (SBCSFieldsReg.sbaccess === 0.U) && (SBCSFieldsReg.sbaccess8   =/= 1.U) ||
+                        (SBCSFieldsReg.sbaccess === 1.U) && (SBCSFieldsReg.sbaccess16  =/= 1.U) ||
+                        (SBCSFieldsReg.sbaccess === 2.U) && (SBCSFieldsReg.sbaccess32  =/= 1.U) ||
+                        (SBCSFieldsReg.sbaccess === 3.U) && (SBCSFieldsReg.sbaccess64  =/= 1.U) ||
+                        (SBCSFieldsReg.sbaccess === 4.U) && (SBCSFieldsReg.sbaccess128 =/= 1.U)
+
+    val compareAddr = Wire(UInt(32.W)) // Need use written or latched address to detect error case depending on how transaction is initiated
+    compareAddr := Mux(SBADDRESSWrEn(0),SBADDRESSWrData(0),SBADDRESSFieldsReg(0))
+
+    val sbAlignmentError = (SBCSFieldsReg.sbaccess === 1.U) && (compareAddr(0)   =/= 0.U) ||
+                           (SBCSFieldsReg.sbaccess === 2.U) && (compareAddr(1,0) =/= 0.U) ||
+                           (SBCSFieldsReg.sbaccess === 3.U) && (compareAddr(2,0) =/= 0.U) ||
+                           (SBCSFieldsReg.sbaccess === 4.U) && (compareAddr(3,0) =/= 0.U)
+
+    sbAccessError.suggestName("sbAccessError")
+    sbAlignmentError.suggestName("sbAlignmentError")
+
+    sb2tl.module.io.wrEn     := tryWrEn && !SBCSFieldsReg.sbbusy && !SBCSFieldsReg.sberror && !sbAccessError && !sbAlignmentError 
+    sb2tl.module.io.rdEn     := tryRdEn && !SBCSFieldsReg.sbbusy && !SBCSFieldsReg.sberror && !sbAccessError && !sbAlignmentError
+    sb2tl.module.io.sizeIn   := SBCSFieldsReg.sbaccess
+    sb2tl.module.io.dmactive := dmactive
+
+    when (~dmactive) {
+      SBCSFieldsReg := SBCSFieldsRegReset
+    }.otherwise {
+      SBCSFieldsReg.sbbusyerror     := Mux(SBCSWrEn && SBCSWrData.sbbusyerror,                   false.B, // W1C
+                                       Mux(anyAddressWrEn && SBCSFieldsReg.sbbusy,               true.B, // Set if a write to SBADDRESS occurs while busy
+                                       Mux((anyDataRdEn || anyDataWrEn) && SBCSFieldsReg.sbbusy, true.B, SBCSFieldsReg.sbbusyerror))) // Set if any access to SBDATA occurs while busy
+      SBCSFieldsReg.sbreadonaddr    := Mux(SBCSWrEn, SBCSWrData.sbreadonaddr   , SBCSFieldsReg.sbreadonaddr)
+      SBCSFieldsReg.sbautoincrement := Mux(SBCSWrEn, SBCSWrData.sbautoincrement, SBCSFieldsReg.sbautoincrement)
```

Comment:
Nothing in the spec prevents this explicitly. The only constraint on writing SBCS is: "To avoid race conditions, debuggers must not try to clear `sberror` until they read `sbbusy` as 0.'

Reply:
Ok. I'll file an issue on the spec that clarifies more things the debugger should not try to do (shouldn't change what your implementation has to do).